cai_causal_graph/causal_graph.py

```diff
@@ -741,6 +741,9 @@ def _set_edge(self, edge: Edge, validate: bool = True):
             edge.destination
         )
 
```

Comment:
Would recommend adding an in-line comment here so that others know what's going on, it's not necessary but good practice

Something along the lines of "# raise an error if the reverse edge has already been defined"